### PR TITLE
Remove string usages of old transport settings

### PR DESCRIPTION
**Pull request by Tim-Brooks**
This is related to #36652. We intend to deprecate a number of transport
settings in 7.x and remove them in 8.0. This commit removes the string
usages of these settings.

### CODE REVIEW

**Comment by tbrooks8** [comment]
This commit will go to 7.1 and 8.0. There should not be any backwards compatibility issues because the new settings are on 6.7.

File: buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java [hunk 1]
```diff
@@ -537,7 +537,7 @@ private void createConfiguration()  {
         defaultConfig.put("node.attr.testattr", "test");
         defaultConfig.put("node.portsfile", "true");
         defaultConfig.put("http.port", "0");
-        defaultConfig.put("transport.tcp.port", "0");
+        defaultConfig.put("transport.port", "0");
```
Comment (ywelsch) [inline buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java]: do we need to conditionally set this based on version as well?

File: client/sniffer/src/test/resources/create_test_nodes_info.bash [hunk 2]
```diff
@@ -60,7 +60,7 @@ node${attr}.dummy:  everyone_has_me
 node${attr}.number: ${node:1}
 node${attr}.array:  [${node:0:1}, ${node:1}]
 http.port:          ${http_port}
-transport.tcp.port: ${transport_port}
+transport.port: ${transport_port}
```
Comment (ywelsch) [inline client/sniffer/src/test/resources/create_test_nodes_info.bash]: this  script runs old versions of elasticsearch (2.0.0, 5.0.0, 6.0.0) so should not be updated

**Review by ywelsch — CHANGES_REQUESTED** [review]
Thanks for the clean-up. I've left 2 comments.

**Review by ywelsch — APPROVED** [review]
LGTM

**Pull request merged**